Source/WebCore/page/ImageAnalysisQueue.cpp

```diff
@@ -158,10 +161,26 @@ void ImageAnalysisQueue::resumeProcessing()
                 m_pendingRequestCount--;
 
             resumeProcessingSoon();
+
+            if (m_queue.isEmpty() && m_imageQueueEmptyHysteresis)
+                m_imageQueueEmptyHysteresis->impulse();
         });
     }
 }
 
+void ImageAnalysisQueue::setImageQueueEmptyHysteresis(Function<void()>&& callback)
```

Comment:
Since this just takes a callback now, I'd rename this to `setDidBecomeEmptyCallback`.